controllers/userController.js

```diff
@@ -52,6 +52,41 @@ module.exports = {
       });
   },
 
+  renderEdit: (req, res, next) => {
+    let userId = req.params.id;
+    User.findById(userId)
+      .then(user => {
+        res.render('user/edit', {
+          user: user
+        });
+      })
+      .catch(error => {
+        console.log(`Error fetching subscriber by ID:${error.message}`);
+        next(error);
+      });
+  },
+
+  update: (req, res, next) => {
+    let userId = req.params.id,
+      userParams = {
+        username: req.body.username,
+        email: req.body.email,
+        password: req.body.password
+      };
+    User.findByIdAndUpdate(userId, {
+      $set: userParams
+    })
+      .then(user => {
+        res.locals.redirect = `/user/${userId}`;
+        res.locals.user = user;
+        next();
+      })
+      .catch(error => {
+        console.log(`Error updating subscriber by ID:${error.message}`);
```

Comment:
same here

Reply:
sorry was copy pasting from the book lol